src/meta_memcache/base/connection_pool.py

```diff
@@ -115,13 +115,19 @@ def get_connection(self) -> Generator[MemcacheSocket, None, None]:
         try:
             conn = self._pool.popleft()
         except IndexError:
+            conn = None
+
+        if conn is None:
             conn = self._create_connection()
 
         try:
             yield conn
-        except Exception:
+        except Exception as e:
             # Errors, assume connection is in bad state
-            _log.exception("Error during cache conn context (discarding connection)")
+            _log.warning(
+                "Error during cache conn context (discarding connection)",
+                exc_info=e,
```

Comment:
Is this any different than doing `exc_info=True`?

Reply:
setting it to true will make the logger call `sys.get_exc_info()`, it is basically the same thing, but I don't have strong preference either way. Do we prefer =True?

Reply:
We historically just do `exc_info=True`, but I don't think anyone has a super strong opinion.